src/core/Akka.Streams.Tests/Dsl/FlowThrottleSpec.cs

```diff
@@ -308,7 +309,7 @@ public void Throttle_for_single_cost_elements_must_properly_combine_shape_and_th
                     .Throttle(1, TimeSpan.FromMilliseconds(100), 5, ThrottleMode.Enforcing)
                     .RunWith(this.SinkProbe<int>(), Materializer)
                     .Request(5)
-                    .ExpectNext(1, 2, 3, 4, 5)
+                    .ExpectNext(CancellationToken.None, 1, 2, 3, 4, 5)
```

Comment:
@Arkatufus Maybe we need to add an overload to `ExpectNext` that implicitly sucks up a timeout. Passing in `CancellationToken.None` everywhere feels like code smell.

Reply:
The obsolete one is the one that does that. Maye we should remove the Obsolete attribute instead.

Reply:
Note that this can be turned into an async call instead, it has a slightly different API.
```
await probe.AsyncBuilder()
    .ExpectNext("a-1", "a-2")
    .ExpectNoMsg(TimeSpan.FromMilliseconds(500))
    .ExecuteAsync();
```

Add the `.AsyncBuilder()` at front, then call `.ExecuteAsync()` at the end.

Reply:
> Maybe we should remove the Obsolete attribute instead.

I think that's a good idea - @eaba can you implement that on this PR instead?